dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -38,7 +40,7 @@ def new_general
         return render :no_attendance unless session.attendances.exists?(teacher: current_user)
       end
 
-      @form_id = WorkshopDailySurvey.get_form_id_for_day day
+      @form_id = WorkshopDailySurvey.get_form_id_for_day Pd::WorkshopDailySurvey::CATEGORY_MAP[workshop.subject], day
```

Comment:
This seems cumbersome. It appears the `CATEGORY_MAP` is only used to retrieve a value and pass it right back to another method (`get_form_id_for_day`) on same class. I think it would be cleaner to keep all that logic contained in `Pd::WorkshopDailySurvey` and modify `get_form_id_for_day` to take a subject and day.

Same below

Reply:
That seems fair